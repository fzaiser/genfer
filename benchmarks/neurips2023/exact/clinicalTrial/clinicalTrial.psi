// Clinical trial data set, using 100 of 1000 data points
def main() {
    isEffective := flip(1/2);
    probControl := 0;
    probTreated := 0;
    probAll := 0;
    if isEffective {
        probControl = uniform(0, 1);
        probTreated = uniform(0, 1);
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));
        observe(0 = flip(probControl));
        observe(1 = flip(probControl));

        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(1 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
        observe(0 = flip(probTreated));
    } else {
        probAll = uniform(0, 1);
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));

        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(1 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
        observe(0 = flip(probAll));
    }
    return isEffective;
}
