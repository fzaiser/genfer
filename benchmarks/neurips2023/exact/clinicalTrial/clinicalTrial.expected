# The following occurs with e-1 at the end
1.703615
16111502227443735727250427761812942600405237723/94572409612368043294199619316675018741649913883
